freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -74,33 +77,39 @@
         }, LOGGER, ":::Auto sync::: provider is checked in {}ms");
     }
 
-    private InstanceStatus updateStatuses(CloudVmInstanceStatus vmInstanceStatus, InstanceMetaData instanceMetaData) {
+    private InstanceStatus updateStatuses(CloudVmInstanceStatus vmInstanceStatus, InstanceMetaData instanceMetaData,
+        Map<InstanceMetaData, DetailedStackStatus> instanceHealthStatusMap) {
         LOGGER.info(":::Auto sync::: {} instance metadata status update in progress, new status: {}",
                 instanceMetaData.getShortHostname(), vmInstanceStatus);
         InstanceStatus status = null;
-        switch (vmInstanceStatus.getStatus()) {
-            case STARTED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.CREATED);
-                status = InstanceStatus.CREATED;
-                break;
-            case STOPPED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.STOPPED);
-                status = InstanceStatus.STOPPED;
-                break;
-            case FAILED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.FAILED);
-                status = InstanceStatus.FAILED;
-                break;
-            case TERMINATED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_ON_PROVIDER_SIDE);
-                status = InstanceStatus.DELETED_ON_PROVIDER_SIDE;
-                break;
-            case TERMINATED_BY_PROVIDER:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_BY_PROVIDER);
-                status = InstanceStatus.DELETED_BY_PROVIDER;
-                break;
-            default:
-                LOGGER.info(":::Auto sync::: the '{}' status is not converted", vmInstanceStatus.getStatus());
+        if (DetailedStackStatus.UNREACHABLE == instanceHealthStatusMap.get(instanceMetaData)) {
```

Comment:
I think this might only work for cases where the status is unreachable. For example, if the instance health is unhealthy I suspect the UI will report "running" since it will rely on the VM status which will be started which means the instance status will be created which means the UI will report "running".

Reply:
Thanks for the suggestion! I added an UNHEALTHY status to the instancestatus enum to catch UNHEALTHY separately. If this is merged, I will also add the UNHEALTHY to cloudbreak-cloud UI repository to be able to display UNHEALTHY in UI (https://github.com/hortonworks/hortonworks-cloud/blob/master/web/cloudbreak-ui/src/app/helpers/freeipa.helpers.ts).